### PR TITLE
Synch this code from @bentley/reality-data package.

### PR DESCRIPTION
**Pull request by MarcBedard8**
Found out that the CRS 9300 is not defined in the CRS registry database, so we cannot use CRSManager: https://ogtcrsregistry.blob.core.windows.net/registry/crs/9300.prj We will check for isGeographicCRS and isProjectedCRS and both are going to return false in that case and we will fallback to use un-georeferenced code path.

### CODE REVIEW

File: core/frontend/src/tile/OPCFormatInterpreter.ts [hunk 1]
```diff
@@ -54,9 +54,16 @@ export class OPCFormatInterpreter  {
     worldRange = Range3d.createXYZXYZ(bounds.getMinX(), bounds.getMinY(), bounds.getMinZ(), bounds.getMaxX(), bounds.getMaxY(), bounds.getMaxZ());
     isGeolocated = false;
     const fileCrs = fileReader.getFileCRS();
-    if (fileCrs) {
+    // Fix https://dev.azure.com/bentleycs/SEG/_workitems/edit/1491612
```
Comment (aruniverse) [inline core/frontend/src/tile/OPCFormatInterpreter.ts:57]:
```suggestion
```

File: test-apps/imodel-from-orbitgt/src/OrbitGtContextModelCreator.ts [hunk 2]
```diff
@@ -66,8 +66,15 @@ export class OrbitGtContextIModelCreator {
       const bounds = fileReader.getFileBounds();
       let worldRange = Range3d.createXYZXYZ(bounds.getMinX(), bounds.getMinY(), bounds.getMinZ(), bounds.getMaxX(), bounds.getMaxY(), bounds.getMaxZ());
       let geoLocated = false;
-      if (fileCrs.length > 0) {
-        await CRSManager.ENGINE.prepareForArea(fileCrs, bounds);
+      // Fix https://dev.azure.com/bentleycs/SEG/_workitems/edit/1491612
```
Comment (pmconne) [inline test-apps/imodel-from-orbitgt/src/OrbitGtContextModelCreator.ts:69]:
```suggestion
```

File: core/frontend/src/tile/OPCFormatInterpreter.ts [hunk 3]
```diff
@@ -54,9 +54,15 @@ export class OPCFormatInterpreter  {
     worldRange = Range3d.createXYZXYZ(bounds.getMinX(), bounds.getMinY(), bounds.getMinZ(), bounds.getMaxX(), bounds.getMaxY(), bounds.getMaxZ());
     isGeolocated = false;
     const fileCrs = fileReader.getFileCRS();
-    if (fileCrs) {
+    // the CRS 9300 is not defined in the CRS registry database, so we cannot use CRSManager
+    // Check to isGeographicCRS and isProjectedCRS are both going to return false in that case and we will fallback to
+    // use un-georeferenced code path.
+    await CRSManager.ENGINE.prepareForArea(fileCrs, bounds);
+    // const _isGeoCentric = CRSManager.ENGINE.isGeocentricCRS(fileCrs);
```
Comment (pmconne) [inline core/frontend/src/tile/OPCFormatInterpreter.ts:61]:
```suggestion
```

File: test-apps/imodel-from-orbitgt/src/OrbitGtContextModelCreator.ts [hunk 4]
```diff
@@ -66,8 +66,14 @@ export class OrbitGtContextIModelCreator {
       const bounds = fileReader.getFileBounds();
       let worldRange = Range3d.createXYZXYZ(bounds.getMinX(), bounds.getMinY(), bounds.getMinZ(), bounds.getMaxX(), bounds.getMaxY(), bounds.getMaxZ());
       let geoLocated = false;
-      if (fileCrs.length > 0) {
-        await CRSManager.ENGINE.prepareForArea(fileCrs, bounds);
+      // the CRS 9300 is not defined in the CRS registry database, so we cannot use CRSManager
+      // Check to isGeographicCRS and isProjectedCRS are both going to return false in that case and we will fallback to
+      // use un-georeferenced code path.
+      await CRSManager.ENGINE.prepareForArea(fileCrs, bounds);
+      // const _isGeoCentric = CRSManager.ENGINE.isGeocentricCRS(fileCrs);
```
Comment (pmconne) [inline test-apps/imodel-from-orbitgt/src/OrbitGtContextModelCreator.ts:73]:
```suggestion
```

File: core/frontend/src/tile/OPCFormatInterpreter.ts [hunk 5]
```diff
@@ -54,9 +54,14 @@ export class OPCFormatInterpreter  {
     worldRange = Range3d.createXYZXYZ(bounds.getMinX(), bounds.getMinY(), bounds.getMinZ(), bounds.getMaxX(), bounds.getMaxY(), bounds.getMaxZ());
     isGeolocated = false;
     const fileCrs = fileReader.getFileCRS();
-    if (fileCrs) {
+    // the CRS 9300 is not defined in the CRS registry database, so we cannot use CRSManager
+    // Check to isGeographicCRS and isProjectedCRS are both going to return false in that case and we will fallback to
+    // use un-georeferenced code path.
+    await CRSManager.ENGINE.prepareForArea(fileCrs, bounds);
+    const isGeographicCRS = CRSManager.ENGINE.isGeographicCRS(fileCrs);
+    const isProjectedCRS = CRSManager.ENGINE.isProjectedCRS(fileCrs);
```
Comment (pmconne) [inline core/frontend/src/tile/OPCFormatInterpreter.ts:62]: Is `isProjectedCRS` a trivial check, or should you avoid checking it if `isGeographicCRS` is `true`?

Comment (MarcBedard8) [reply core/frontend/src/tile/OPCFormatInterpreter.ts:62]: That is trivial.

**Review by pmconne — APPROVED** [review]

**Pull request merged**